### PR TITLE
embObjMais: expose data via MAS interface

### LINKED ISSUE #851: Expose information provided by embObjMAIS YARP device using a non-deprecated interface
### Is your feature request related to a problem?

I noticed that the `embObjMAIS` device is currently exposing its information via the `IAnalogSensor` YARP interface, that anyhow has been deprecated (see https://www.yarp.it/latest/group__dev__impl__deprecated.html). 

Probably we could consider to expose the info via either:
* [`yarp::dev::IEncoderArrays`](https://www.yarp.it/latest/classyarp_1_1dev_1_1IEncoderArrays.html): interface that is part of the multiple analog sensor interfaces, however has the downside that you cannot expose a name for each encoder
* [`yarp::dev::IAxisInfo`](https://www.yarp.it/latest/classyarp_1_1dev_1_1IAxisInfo.html) and [`yarp::dev::IEncoders`](https://www.yarp.it/latest/classyarp_1_1dev_1_1IEncoders.html) these interfaces are the same used to expose "regular"  joint encoders, but I am not sure that the `nws`  and `nwc`  work fine if one implements only these interfaces and not all the rest of control-related interfaces.

### The solution you would like to have available

See feature description.

### Alternatives you have considered

See feature description.

### Additional context

I realized this while thinking about https://github.com/orgs/robotology/discussions/609 .

### PR DESCRIPTION
**Pull request by Nicogene**
It fixes #851.

Not sure if this implementation makes sense @marcoaccame @traversaro.
If it is ok I will test it on `iCubGenova11`

### CODE REVIEW

File: src/libraries/icubmod/embObjMais/embObjMais.cpp [hunk 1]
```diff
@@ -530,14 +530,39 @@ int embObjMais::calibrateChannel(int ch, double v)
 }
 
 
+size_t embObjMais::getNrOfEncoderArrays() const {
+    return 1;
+}
+
+yarp::dev::MAS_status embObjMais::getEncoderArrayStatus(size_t sens_index) const {
```
Comment (traversaro) [inline src/libraries/icubmod/embObjMais/embObjMais.cpp:537]:
```suggestion
yarp::dev::MAS_status embObjMais::getEncoderArrayStatus(size_t sens_index) const {
if (sens_index >= 1) return yarp::dev::MAS_UNKNOWN;
```

File: src/libraries/icubmod/embObjMais/embObjMais.cpp [hunk 2]
```diff
@@ -530,14 +530,39 @@ int embObjMais::calibrateChannel(int ch, double v)
...
+    return yarp::dev::MAS_OK;
+}
+
+bool embObjMais::getEncoderArrayName(size_t sens_index, std::string &name) const {
```
Comment (traversaro) [inline src/libraries/icubmod/embObjMais/embObjMais.cpp:541]:
```suggestion
bool embObjMais::getEncoderArrayName(size_t sens_index, std::string &name) const {
    if (sens_index >= 1) return false;
```

File: src/libraries/icubmod/embObjMais/embObjMais.cpp [hunk 3]
```diff
@@ -530,14 +530,39 @@ int embObjMais::calibrateChannel(int ch, double v)
...
+    name = serviceConfig.nameOfMais;
+    return true;
+}
+
+bool embObjMais::getEncoderArrayMeasure(size_t sens_index, yarp::sig::Vector& out, double& timestamp) const {
```
Comment (traversaro) [inline src/libraries/icubmod/embObjMais/embObjMais.cpp:546]:
```suggestion
bool embObjMais::getEncoderArrayMeasure(size_t sens_index, yarp::sig::Vector& out, double& timestamp) const {
    if (sens_index >= 1) return false;
```

File: src/libraries/icubmod/embObjMais/embObjMais.cpp [hunk 4]
```diff
@@ -530,14 +530,39 @@ int embObjMais::calibrateChannel(int ch, double v)
...
+    timestamp = this->timeStamp;
+    out.resize(analogdata.size());
+    out = analogdata;
+    return true;
+}
+
+size_t embObjMais::getEncoderArraySize(size_t sens_index) const {
```
Comment (traversaro) [inline src/libraries/icubmod/embObjMais/embObjMais.cpp:553]:
```suggestion
size_t embObjMais::getEncoderArraySize(size_t sens_index) const {
    if (sens_index >= 1) return 0;
```

**Review by traversaro — APPROVED** [review]
Minor comments.

**Review by pattacini — APPROVED** [review]

**Pull request merged**